app/jobs/regular/bump_topic.rb

```diff
@@ -5,18 +5,19 @@ class BumpTopic < ::Jobs::Base
 
     def execute(args)
       topic_timer = TopicTimer.find_by(id: args[:topic_timer_id])
+      return if !topic_timer&.runnable?
 
-      topic = topic_timer&.topic
-
-      if topic_timer.blank? || topic.blank? || topic_timer.execute_at > Time.zone.now
+      topic = topic_timer.topic
+      if topic.blank?
+        topic_timer.destroy!
         return
       end
```

Comment:
The new changes are duplicated across the jobs. I think we can create a parent job and have 'TopicTimer' jobs subclass the parent class.

Reply:
Yes good idea, they were not very consistent before. Will make a base class. Topic timers are not going away, they are just not going to be scheduled jobs anymore.